File: examples/libs/bmp/lv_example_bmp_1.py
#!/opt/bin/lv_micropython -i
import lvgl as lv
import display_driver
import fs_driver

fs_drv = lv.fs_drv_t()
fs_driver.fs_register(fs_drv, 'S')

img = lv.img(lv.scr_act())
# The File system is attached to letter 'S'

img.set_src("S:example_32bit.bmp")
img.center()

File: examples/libs/bmp/lv_example_bmp_1.c
#include "../../lv_examples.h"
#if LV_USE_BMP && LV_BUILD_EXAMPLES

/**
 * @title Decode BMP from filesystem
 * @brief Load a 32-bit BMP file through the LVGL filesystem and center it on the screen.
 *
 * An image widget is created on the active screen and its source is set to
 * `A:lvgl/examples/libs/bmp/example_32bit.bmp` so the BMP decoder reads the
 * file through an attached filesystem driver registered under drive letter
 * `A`, such as `LV_USE_FS_STDIO`.
 */
void lv_example_bmp_1(void)
{
    lv_obj_t * img = lv_image_create(lv_screen_active());
    /* Assuming a File system is attached to letter 'A'
     * E.g. set LV_USE_FS_STDIO 'A' in lv_conf.h */
    lv_image_set_src(img, "A:lvgl/examples/libs/bmp/example_32bit.bmp");
    lv_obj_center(img);

}

#endif
